Code Review pull request

apesic opened the pull request.

Commit:
Modified seed file

Commit:
Modified seed file

Commit:
Fixed seed file formatting error

Commit:
Fixed sentence parent id to use id rather than position.

- Made initial render handle first three sentences before rendering
  additional columns.

Commit:
Merge pull request #48 from apesic/js-bugfix

Js bugfix

jbwhite committed:
add delete route, js, and controller for deleting exported files from tmp

jbwhite committed:
Merge branch 'master' into debugExport

Commit:
Fixed child sentences not receiving parent id, renamed updateCue to updateChildren to reflect this change

Commit:
Merge pull request #49 from apesic/parent-id-bug

Fixed child sentences not receiving parent id

Commit:
Add spacing between families on stories show page

jbwhite committed:
Merge branch 'master' into debugExport

Commit:
Add consistent spacing between columns

Commit:
Merge pull request #50 from woolly-bear-caterpillars-2014/debugExport

Debug export, add delete method, controller, and call for exported stories

Commit:
Give first column its own class, firstthree

stevecass commented on README.rdoc:
```diff
@@ -0,0 +1,28 @@
+== README
```
You can delete this readme as you have readme.md

stevecass commented on app/assets/javascripts/horizontal.js:
```diff
@@ -0,0 +1,149 @@
+    Mousetrap.bind('right', function() {
+        screenRight = $(document).scrollLeft();
+        $("html, body").animate({scrollLeft: (screenRight + $(".column").width())}, 300);
+        // currentPos = (currentPos + 200);
```
It looks like 200 and 300 have some significance. Think about declaring constants rather than using a "magic number."

stevecass commented on app/assets/javascripts/horizontal.js:
```diff
@@ -0,0 +1,149 @@
+
+
+
+// $(document).keydown(function(e) {
```
It's best not to have commented out code in master. Also, remove these multiple blank lines.

stevecass commented on app/assets/javascripts/jquery.fileDownload.js:
```diff
@@ -0,0 +1,453 @@
+/*
```
Think about putting downloaded/library js files in a "vendor" folder to keep it separate from your own code.

stevecass commented on app/assets/javascripts/horizontal.js:
```diff
@@ -0,0 +1,149 @@
+
+    Mousetrap.bind(['shift+right', 'shift+4'], function() {
+        $("html, body").animate({scrollLeft: $(document).width() }, 300);
+        // currentPos = (currentPos + 200);
```
Remove these commented-out lines.

stevecass commented on app/helpers/sentences_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module SentencesHelper
```
You should remove empty helpers created by generators. You can always recreate them if you need them.

stevecass commented on app/assets/javascripts/stories.js:
```diff
@@ -0,0 +1,271 @@
+};
+
+Sentence.prototype.updateChildren = function() {
+  var cue = $('.cue[data-parent-position=' + this.position + ']');
```
Given that you have a view module, it might make sense to move some of the sentence-display logic into it (or into a sentence viewing module), and keep the sentence modules concerned with how sentences are loaded, saved and organized into hierarchies. This DOM manipulation code feels out of place in a module labeled "Sentence model."

stevecass commented on app/assets/stylesheets/horizontal.css:
```diff
@@ -0,0 +1,194 @@
+    width: 0;
+    height: 0;
+    line-height: 0;
+    border: 0.5921052631578947em solid transparent;
```
What is the significance of this number? Should it be a sass variable / explained in some way?

stevecass commented on app/assets/stylesheets/horizontal.css:
```diff
@@ -0,0 +1,194 @@
+	background: #F5FEFF;
+    /*border: solid black;*/
+    /*border-radius: 20px;*/
+	color: #ddd;
```
Consider using sass and putting these definitions in your _color_variables.css

stevecass commented on app/views/layouts/application.html.erb:
```diff
@@ -0,0 +1,27 @@
+<html>
+<head>
+  <title>three sentences</title>
+  <script src="http://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.6.0/underscore-min.js"></script>
```
Download this and add to assets -  allow the asset pipeline to minimize the number of web requests you make.

stevecass commented on app/views/stories/test.html.erb:
```diff
@@ -0,0 +1,497 @@
+<!DOCTYPE html>
```
Does this file need to exist on master?

stevecass commented on app/views/welcome/index.html.erb:
```diff
@@ -0,0 +1,38 @@
+      <div class="grid">
+      <h2><%= @user.username %>'s stories</h2><br><br>
+      <li><%= link_to 'New Story', new_story_path, class: 'button newstory' %></li>
+        <ol>
```
Think about spacing, and be consistent.

stevecass replied:
This OL tag is superfluous?

stevecass commented on app/views/welcome/index.html.erb:
```diff
@@ -0,0 +1,38 @@
+
+           <div class="grouping">
+            <div class="col-1">
+            <i class="my-icon"><i></i><i></i><i></i></i><h4><%= story.sentences.size %></h4><br><h6><%= pluralize(story.sentences.size, 'sentence')%><h6><h5><%= link_to story.name, story %></h5>
```
What's going on here with the empty i tags? This file might need tidying up.

stevecass commented on public/test.json:
```diff
@@ -0,0 +1,399 @@
+{
```
Does this belong on master?

stevecass commented on app/models/story.rb:
```diff
@@ -0,0 +1,83 @@
+
+  def get_content(depth)
+    md_content = ""
+    md_content += create_title
```
You can combine this line and the line above for readabilitiy

stevecass commented on app/models/story.rb:
```diff
@@ -0,0 +1,83 @@
+      end
+    else
+      headers = get_headers(depth - 1)
+      headers
```
Not clear what the intention of this line is? Was this intended to be a return value? No, as not last line evaluated. What then?

stevecass commented on app/controllers/sentences_controller.rb:
```diff
@@ -0,0 +1,34 @@
+    if @sentence.save
+      render json: @sentence.to_json
+    else
+      render json: @sentence.errors.full_messages.to_json
```
Handling save errors is good, but what does the front end do with this json? Set a status code too so your  front-end can call the failure handler rather than the success handler.

Commit:
Add spacing between families and columns

Commit:
Merge pull request #53 from postmodem/spacing

Add spacing between families and columns

stevecass commented on README.md:
```diff
@@ -1,4 +1,30 @@
-threeSentences
+Three Sentences ![travis-ci-status](https://travis-ci.org/woolly-bear-caterpillars-2014/threeSentences.svg?branch=master)
```
Also add a contributing file outlining the group workflow for the project.

stevecass commented on app/assets/javascripts/stories.js:
```diff
@@ -0,0 +1,271 @@
+  var columnTemplate,
+      cueTemplate,
+      sentenceTemplate,
+      currentSentenceContent;
```
Consider putting these variables in a namespace. For example if you had 
`var moduleState = {};` your later references could be along the lines of `moduleState.currentSentence = blah;` This would make it easier for the reader to distinguish between the state of the view and local variables inside single functions.

jbwhite committed:
update export function to include all children

stevecass commented on app/assets/javascripts/export.js:
```diff
@@ -0,0 +1,32 @@
+
+
+  });
+  $('#exportform form').submit(function(e){
```
Be consistent with blank lines. Add at least one blank above here, but remove the extra blanks above the closing brace of the previous block.

jbwhite committed:
update export functions

jbwhite committed:
Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences

jbwhite committed:
Merge branch 'master' into exportFormat

stevecass commented on app/assets/javascripts/stories.js:
```diff
@@ -0,0 +1,279 @@
+
+// ----- SENTENCE MODEL ------
+var Sentence = function(sentenceJson) {
+  this.id = sentenceJson.id;
```
Check what your intent was here. Unlike your view model definition you are not inside a closure here.

jbwhite committed:
update export form and add shortened_title method

jbwhite committed:
create punctuate method

Commit:
Horizontal scrolling - dynamically set body with

Commit:
Add custom scrollbars and tweak spacing on columns

postmodem committed:
Merge pull request #54 from postmodem/custom-scrollbars

Add custom scrollbars and tweak spacing on columns

Commit:
Removed unneeded jquery plugins

jbwhite committed:
add cancel option to export form

jbwhite committed:
Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences

Commit:
Merge pull request #60 from apesic/horiz-scrolling

Horiz scrolling

jbwhite committed:
Merge branch 'master' into exportFormat

jbwhite committed:
clarify file_line variable name

Commit:
Removed test files and fixed formatting errors

Commit:
Merge pull request #61 from woolly-bear-caterpillars-2014/exportFormat

Export format

Commit:
Added installation info to README

Commit:
Fixed typo in readme

Commit:
Merge pull request #63 from apesic/cleanup

Cleanup - removed test files and commented code

Commit:
Fixed another typo in README

Commit:
Edit styling for stories index page, hover links over icons

Commit:
Fix merge conflict

Commit:
Merge pull request #65 from jwag789/master

hover links on icons in index

Commit:
Turn signed_out into an empty string to fix error message bug on Sign Up

jbwhite committed:
set zindex on absolute-center

jbwhite committed:
Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences

Commit:
Merge pull request #66 from postmodem/error-bug

Turn signed_out into an empty string to fix error message bug on Sign Up

Commit:
Removed mCustomScrollbar plugin, too many bugs

Commit:
Removed mCustomScrollbar plugin, too many bugs

Commit:
Fixed merge conflict

Commit:
Merge pull request #69 from apesic/horiz-scrolling

Removed mCustomScrollbar plugin

jbwhite committed:
Merge pull request #68 from woolly-bear-caterpillars-2014/exportFix

Export fix

jbwhite committed:
reduce font size, adjust spacing of first 3 sentences

jbwhite committed:
Merge branch 'master' into view

Commit:
Merge pull request #70 from woolly-bear-caterpillars-2014/view

View -- adjust 1st column layout

Commit:
Removed horizontal scrollbar for column divs

Commit:
Removed background from scrollbar

Commit:
Added end condition at depth >= 5

- Won't add new columns past 5
- Display message to user that they're done
- Display link to export on end column
- Added fontawesome

Commit:
Add basic tour implementation and tour button

Commit:
End column export button working

- made end column hidden div
- Removed commented code from stylesheets

Commit:
Add tentative steps for tour

Commit:
Restyled nav, added icons.

Removed unnecessary css from styleguide.

Commit:
Merge pull request #72 from apesic/horiz-scrolling

Added end state at depth >= 5, started nav bar restyle

Commit:
Add styling for tooltips

Commit:
Fix merge conflicts and merge tour guide with help button

Commit:
Merge pull request #73 from jwag789/tour_guide

Help button integrated with tour guide

Commit:
Added update method for Stories Controller

jbwhite committed:
setup basic share controller, view, route, and js

jbwhite committed:
resolve merge conflict in stories.js

Commit:
Change copy from book to story to be less specific

Commit:
Add margin to end-message h3 to give it some space

Commit:
Change tour font to #333333 Varela Round and increase line height

postmodem committed:
Merge pull request #75 from postmodem/master

Change wording and spacing on end-message and font and line-height on tour

Commit:
Merge pull request #74 from woolly-bear-caterpillars-2014/share

Share -- basic setup for share function

Commit:
Delete share view

Commit:
Add share_url column to stories table

Commit:
Add before_save and method to generate and save share_url

Commit:
Remove Digest method from StoriesController

Commit:
Fix show.html.erb for sharing

Commit:
Add to_s to random number with generate_share_url so Digest::SHA1 works

Commit:
Added story title editing

Commit:
Merge pull request #76 from apesic/polishing-edit-view

Added story title editing

Commit:
Implement generation/save of share_url on Story creation

Commit:
Set up route for share_url

Commit:
Remove already_signed_out message from devise config

Commit:
Fix error for shared being undefined, implement introjs programatically

Commit:
Fix error for shared being undefined, implement introjs programatically

Commit:
Fix error so hash doesnt show when you click tour button, add tentative demo route

Commit:
Fix error so hash doesnt show when you click tour button, add tentative demo route

Commit:
Create share partial and render in stories#show

Commit:
Add share functionality to export modal

postmodem committed:
Merge pull request #77 from postmodem/share

Share

Commit:
Fix merge conflicts

postmodem committed:
Merge pull request #78 from postmodem/share

Pull upstream and fix merge conflicts

Commit:
Add redirect on done and skip buttons on tour, configure tour routes

Commit:
Add redirect on done and skip buttons on tour, configure tour routes

Commit:
Fix merge conflicts

Commit:
Add share to export modal and style it

Commit:
Add pointer for cue click

Commit:
Add pointer for cue click

Commit:
Clean up and improve export/share modal

postmodem committed:
Merge pull request #80 from postmodem/share

Share

postmodem committed:
Merge pull request #81 from woolly-bear-caterpillars-2014/share

Share

Commit:
Remove share from welcome page

postmodem committed:
Merge pull request #82 from postmodem/share

Remove share from welcome page

Commit:
Fix merge conflicts

Commit:
Merge branch 'jwag789-master'

* jwag789-master: (26 commits)
  Add pointer for cue click
  Add redirect on done and skip buttons on tour, configure tour routes
  Fix error so hash doesnt show when you click tour button, add tentative demo route
  Fix error for shared being undefined, implement introjs programatically
  Clean up and improve export/share modal
  Add share to export modal and style it
  Add share functionality to export modal
  Create share partial and render in stories#show
  Remove already_signed_out message from devise config
  Set up route for share_url
  Implement generation/save of share_url on Story creation
  Added story title editing
  Added update method for Stories Controller
  Add to_s to random number with generate_share_url so Digest::SHA1 works
  Fix show.html.erb for sharing
  Remove Digest method from StoriesController
  Add before_save and method to generate and save share_url
  Add share_url column to stories table
  Delete share view
  Change tour font to #333333 Varela Round and increase line height
  ...

Commit:
Merge pull request #83 from woolly-bear-caterpillars-2014/share

Share

jbwhite committed:
add div.triad around sentences

jbwhite committed:
setup slidetoggle

jbwhite committed:
implement slide toggle

Commit:
Edit tour to include export page and static demo page

jbwhite committed:
Merge pull request #84 from jbwhite/toggle

Toggle

Commit:
Fix merge conflicts

Commit:
Fix merge error

Commit:
Implement autoload for intro tour

Commit:
Merge pull request #85 from jwag789/master

Implement tour

Commit:
Merge branch 'master' of heroku.com:threesentences

* 'master' of heroku.com:threesentences:
  Fixed seed file formatting error
  Modified seed file

Commit:
Edit demo page to use prewritten story

Commit:
Merge pull request #86 from jwag789/master

Edit demo page to use prewritten story

Commit:
Moved nav items into partial and restyled

jbwhite committed:
experiment with scrollspy

jbwhite committed:
scroll

jbwhite committed:
fix triad nesting

jbwhite committed:
removed export/scroll edits

jbwhite committed:
fix conflict

jbwhite committed:
Merge pull request #87 from jbwhite/triad

Triad

Commit:
Implement modal for help

Commit:
Fixed bug with export modal not registering clicks

Commit:
Add spacing between clusters

Commit:
Merge from upstream to fix export modal

Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences

Commit:
Merge pull request #88 from apesic/polishing-edit-view

Polishing view

Commit:
Merge to fix export button and add help modal

Commit:
Added indicator arrows between columns

Commit:
Made column arrows clickable, added title to nav bar and styled.

- Styled 'saved' indicator
- Tweaked column spacing

Commit:
Added link and styling to header tagline

Commit:
Merge pull request #89 from apesic/branch-ornamentation

Added clickable column arrows, added title to nav bar.

Commit:
Tweaked export modal, made share url input read only.

Commit:
Add/style help modal and change focus style on tour buttons

Commit:
Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences

Commit:
Merge pull request #90 from jwag789/master

Create/style help modal and change focus styling on tour buttons

Commit:
Take out redundencies in code on help js file

Commit:
Fix url redirect to work for any window location origin

Commit:
Merge pull request #91 from jwag789/master

Fix url for when intro tour loads automatically

Commit:
Fixed layout of left login buttons, moved logic to view helper

Commit:
Change icons on welcome (home) to font awesome - clean up view too

Commit:
Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences

Commit:
Merge pull request #92 from apesic/branch-ornamentation

Tweaked export modal, made share url input read only.

Commit:
Merge pull request #93 from jwag789/master

Change icons on welcome (home) to font awesome - clean up view too

jbwhite committed:
night mode refactoring

jbwhite committed:
night mode refactoring

jbwhite committed:
Merge branch 'master' into nightmode

Commit:
Merge pull request #94 from jbwhite/nightmode

Nightmode

Commit:
Merge branch 'master' of heroku.com:threesentences

* 'master' of heroku.com:threesentences:
  Fixed seed file formatting error
  Modified seed file

jbwhite committed:
skinzzz

jbwhite committed:
Merge branch 'master' into nightmode

jbwhite committed:
clean up

jbwhite committed:
Merge pull request #101 from jbwhite/skinz

Skinz

Commit:
Fixed bug with parent-id updating

Commit:
Merge pull request #102 from apesic/view-tweaks

Bugfixes

jbwhite committed:
edit export form

jbwhite committed:
update export form to allow clicking on labels to select

Commit:
Merge pull request #104 from jbwhite/export

update export form to allow clicking on labels to select

jbwhite committed:
reduced font size for share url

jbwhite committed:
Merge branch 'master' into export_fontsize

jbwhite committed:
Merge pull request #105 from jbwhite/export_fontsize

Export fontsize

Commit:
Put navbar in front of other layers

Commit:
Merge pull request #106 from jwag789/master

Put navbar in front of other layers

Commit:
Fixed bug with story editing in share mode.

- Found and fixed another share mode bug--url was being generated
  before_save rather than before_create, so it was invalid once
  the story was modified
- Fixed indentation in JS file.

Commit:
Merge pull request #107 from apesic/bugfix

Fixed bug with story editing in share mode.

Commit:
Cull shortened_title method and add gsub and range to sanitize/shorten filenames

postmodem committed:
Merge pull request #108 from postmodem/quotation-export-fix

Cull shortened_title method and add gsub and range to sanitize/shorten f...

Commit:
Fix demo bug so styling works in production mode

Commit:
Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences

Commit:
Merge pull request #109 from jwag789/master

Fix bug for styling in demo page in production mode

Commit:
Change sizing for page icons on index page

jbwhite committed:
set up export module

jbwhite committed:
set up export module

jbwhite committed:
set up export module

Commit:
Added helper to shorten title names on index

Commit:
Merge pull request #110 from jwag789/master

Helper to shorten title names

jbwhite committed:
moved export methods to Export module

jbwhite committed:
set generate_share_url as private method

jbwhite committed:
resolve merge conflict

Commit:
Add disable_with and reset button to Export on form_tag

postmodem committed:
Merge pull request #111 from postmodem/export-spinner

Add disable_with and reset button to Export on form_tag

Commit:
Edit keyboard shortcut, allow some shortcuts on input fields

Commit:
Took out global binding for right keyboard shortcut

Commit:
Add correct nav icons to demo page

Commit:
Merge pull request #112 from jwag789/master

Change keyboard shortcuts

Commit:
Change copy on tour to clarify and make it pithier

postmodem committed:
Merge pull request #113 from postmodem/tighten-up-tour

Change copy on tour to clarify and make it pithier

Commit:
Add scrollLeft to keep tour element in viewport

Commit:
Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences into tighten-up-tour

Commit:
Add arrows to slide toggle

Commit:
Take out console logs for slide toggle arrow implementation

postmodem committed:
Merge pull request #115 from postmodem/tighten-up-tour

Add scrollLeft to keep tour element in viewport

jbwhite committed:
finish refactoring export module

jbwhite committed:
fixed merge conflicts which should not have existed to begin with

jbwhite committed:
Merge pull request #117 from jbwhite/exportModule

Export module

Commit:
Edit arrow for toggle

Commit:
Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences

Commit:
Merge pull request #118 from jwag789/master

Arrows for collapse

Commit:
Edit arrow color for toggle

Commit:
Merge pull request #119 from jwag789/master

Edit arrow color for toggle

jbwhite committed:
add story title to html title if present

jbwhite committed:
Merge pull request #120 from woolly-bear-caterpillars-2014/title

add story title to html title if present

Commit:
Change over to html5 validations and remove previous error handling

postmodem committed:
Merge pull request #121 from postmodem/use-html5-validations-for-error-handling

Change over to html5 validations and remove previous error handling

Commit:
Change copy to be more friendly

postmodem committed:
Merge pull request #122 from postmodem/copy-change

Change copy to be more friendly

Commit:
Fix error on export button being hidden behind tooltip highlight

jbwhite committed:
add css for end class to fix export button position

jbwhite committed:
Merge branch 'master' into exportButton

jbwhite committed:
Merge pull request #123 from jbwhite/exportButton

Export button

Commit:
Change the way help link is made

Commit:
Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences

Commit:
Merge pull request #124 from jwag789/master

Highlight doesn't get in the way of the export button in tour

Commit:
Edit keyboard shortcuts

Commit:
Split index and user page into partials, removed signup from index nav

Commit:
Merge pull request #125 from woolly-bear-caterpillars-2014/layout

Split index and user page into partials, removed signup from index nav

Commit:
Merge and edit keyboard shortcuts

Commit:
Moved signup button css to welcome.css.scss

Commit:
Merge pull request #126 from woolly-bear-caterpillars-2014/layout

Moved signup button css to welcome.css.scss

jbwhite committed:
made signup button disappear

jbwhite committed:
Merge pull request #127 from woolly-bear-caterpillars-2014/signUp

made signup button disappear

Commit:
Precompile assets for heroku

Commit:
Precompile assets for heroku

Commit:
Edit keyboard shortcuts and text on help modal

Commit:
Merge from master

Commit:
Create CONTRIBUTING.md

Commit:
Merge pull request #128 from jwag789/master

Fix keyboard shortcuts

Commit:
Renable Export button on modal reload

postmodem committed:
Merge pull request #129 from postmodem/master

Renable Export button on modal reload

jbwhite committed:
new database seed file

jbwhite committed:
Merge pull request #130 from woolly-bear-caterpillars-2014/seed

new database seed file

Commit:
restyled story list page

Commit:
Moved short title method into story model

Commit:
Restyled login and signup forms, fixed error messages

Commit:
Styled error flash for login form

Commit:
Minor JS refactor

* Moved tour function out of global
* Moved next arrow listener into proper function

Commit:
Style tweaks - add border below clusters

Commit:
Fixed label names on registration form

Commit:
Reverted changes to story json to fix tour page

Commit:
Added favicon

Commit:
Merge pull request #131 from apesic/restyle

Improved error feedback for login/signup, restyled story list

Commit:
Fixed typo in event handlers

Commit:
Merge pull request #132 from apesic/restyle

Fixed typo in event handlers

Commit:
Merge branch 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences

* 'master' of https://github.com/woolly-bear-caterpillars-2014/threeSentences:
  Fixed typo in event handlers

Commit:
Merge branch 'master' of heroku.com:threesentences

* 'master' of heroku.com:threesentences:
  Precompile assets for heroku

Commit:
Recompiled assets for Heroku

Commit:
Manually added font-awesome to try and fix missing font issue

Commit:
Added font-awesome fonts to font folder as well
